### ISSUE TITLE
 [BUG] File name too long to be handled

### ISSUE THREAD

**Issue by marcosrdac** [ncmpcpp/ncmpcpp#380]
# File name too long

When mpd plays a youtube video, all open ncmpcpp instances crash (and cannot be oppened again while that video is being streammed) as it is not able to handle a title so long as the one that's being fed into it.

Error:
```
terminate called after throwing an instance of 'boost::filesystem::filesystem_error'
what():  boost::filesystem::status: File name too long: "/home/marcosrdac/.cache/ncmpcpp/lyrics//videoplaybackexpire=1587969835&ei=yyqmXvuiMZaixgSFh7P4Cw&ip=2804%3A7f7%3Aa78a%3A2223%3Af021%3Ae3c%3A74ad%3Ac7ae&id=o-ADoKdC6bh8ChfPQfgsKOfTDbQpw1Ro_0w2GGNTtC0xt6&itag=251&source=youtube&requiressl=yes&mh=0T&mm=31%2C29&mn=sn-b8u-nn5s%2Csn-gpv7dn7k&ms=au%2Crdu&mv=m&mvi=3&pl=38&initcwndbps=480000&vprv=1&mime=audio%2Fwebm&gir=yes&clen=2759856&dur=158.txt"

zsh: abort (core dumped)  ncmpcpp
```

I perceive it happens because the title in question is a very big url. Anyway, mpd can play it, so we got a problem in our best mpd player client hahaha

```
# add video stream to mpd playlist
youtube-dl --add-metadata --prefer-insecure -g "ytsearch1:rammstein" | mpc add

# play last song in playlist (the stream)
mpc play $(mpc playlist | wc -l)
```

**Expected behavior**
ncmpcpp run normally without crashing.
 
**Desktop**
 - OS: Arch Linux x86_64
 - Terminal: st (running zsh or bash)
 - Version 0.21.11

**Additional context**
I believe this bug is the same reported at #55.

**Comment by lenerd** [comment]
Hi @marcosrdac,

I can reproduce the crash when ncmpcpp is configured to download the lyrics in the background, e.g., with the following setting:
```
fetch_lyrics_for_current_song_in_background = "yes"
```
It looks like ncmpcpp tries to save the lyrics to file and uses the "song title" -- in this case a long part of the url -- as filename. This is too long for the filesystem (e.g., ext4 supports filenames up to 255 bytes).

So a workaround would be to disable automatic fetching of lyrics.

A possible fix would be to hash the song title to some smaller string and use that as filename.

**Comment by marcosrdac** [comment]
Sorry for taking so much to answer you, @lenerd . I'm much appreciated about your workaround. I would not get to the conclusion that ncmpcpp's problem was at getting lyrics for the "song". Anyway, it's still a bug for us to keep one eye at.

**Comment by m040601** [comment]
Just chiming in to report the same issue and try to help pinpoint the problem.

I also believe this is related to #55 x 

My system:
- is also Archlinux
- this happens both on x86 or arm (raspberry pi)
- It doesnt matter which terminal or shell

I strongly suspect it is related to the metadata containing something "too big" or "illegal" for ncmpcpp. Other clients with mpd play the files/url without problem.

The reason of my suspicion is that I've been adding for the last years, adding huge amounts of music "scrapped"  from youtube with youtube-dl.

Typically something like this,

    youtube-dl -f m4a --add-metatada https://youtube...video

I get an XYZ.m4a (aac codec) audio file with this.
The XYZ file name is derived from the name of the video was on youtube. It may contain whatever garbage people might put in there, lots of unicode emoticons, "dangerous" characters etc. But usually my ext4 linux file system copes well with that, and I never bother to cleanup or rename the files.

Another thing that --add-metadata option from youtube-dl does

**Comment by m040601** [comment]
UPDATE from my previous comment.

As a recap, the messages I'm seeing are either

`terminate called after throwing an instance of 'MPD::ClientError'`

or

`Response line too large`

Which I am 99% sure is also related with issue
https://github.com/ncmpcpp/ncmpcpp/issues/55



I noticed this happens not only with ncmpcpp, but also ncmpc and even web based clients like ympd and mympd. In any case

**Comment by m040601** [comment]
Another example, https://www.youtube.com/watch?v=63WIYqACGr4

This one besides containing a huge amount of text, it also an example of files containing some ?problematic ?dangerous characters that end up in a metadata tag:

Notice where it says:

> From ''  Everybody's Talkin' ''
> Label: ATCO Records ‎– SD33-385
> Format: Vinyl, LP, Album, PRC Richmond
> Country: US
> Released: 1972

I noticed in vim that it contains this character, "<200e>", that also ends up in the metadata tag "Comment".

It's right here in this line 

Label: ATCO Records **<200e>** ‎– SD33-385

**Comment by arybczak** [comment]
`youtube-dl --add-metadata --prefer-insecure -g "ytsearch1:rammstein"` doesn't return anything here.

Any other way to reproduce?

**Comment by JaanVr** [comment]
@arybczak 
youtube-dl --add-metadata "https://www.youtube.com/watch?v=f6NsZNp6Hhw"
This video and its metadata causes similar issues on my setup.

**Comment by marcosrdac** [comment]
> `youtube-dl --add-metadata --prefer-insecure -g "ytsearch1:rammstein"` doesn't return anything here.
> 
> Any other way to reproduce?

This youtube-dl command searches youtube for the word rammstein and returns the first video as two urls: one for video and another for audio. Here you can see it:

![image](https://user-images.githubusercontent.com/7348840/104047247-1b635f00-51c0-11eb-8cc7-739e3713f26a.png)

**Comment by arybczak** [comment]
Should be fixed by #561.
